packages/workbox-webpack-plugin/src/lib/get-webpack-manifest-entries.js

```diff
@@ -0,0 +1,76 @@
+const url = require('url');
+
+const entry = (revision, filePath, publicPath) => ({
+  revision,
+  url: url.resolve(publicPath, filePath),
+});
+
+/**
+ * Filter chunks that have one of the provided names (from config.chunks)
+ * @param {Array<Object>} chunks Webpack chunks
+ * @param {Array<string>} include Chunk names to include
+ * @param {Array<string>} exclude Chunk names to exclude
+ * @return {Array<Object>} Filtered array of chunks
+ */
+const filterChunks = (chunks, include, exclude) =>
+  chunks.filter((chunk) =>
+    // chunk has a name
+    Object.prototype.hasOwnProperty.call(chunk, 'name')
+      // did not specify include or chunk in include
+      && (!Array.isArray(include) || !!~include.indexOf(chunk.name))
+      // did not specify exclude or chunk not in exclude
+      && (!Array.isArray(exclude) || !~exclude.indexOf(chunk.name))
+  );
+
+/**
+ * Maps webpack asset filenames to their chunk hash
+ * @param {Array<Object>} chunks Webpack chunks
+ * @return {Object} {filename: hash} map
+ */
+const mapAssetsToChunkHash = (chunks) =>
+  chunks.reduce((assetMap, chunk) =>
+      Object.assign(
+        assetMap,
+        ...chunk.files.map((f) => ({[f]: chunk.renderedHash}))
+    ), {});
+
+const generateManifestEntries = (compiler, compilation, config) => {
+  const {hash} = compilation; // compilation hash
+  const {publicPath} = compilation.options.output;
+
+  // Array<string> | undefined
+  const includeNames = config.chunks;
+  const excludeNames = config.excludeChunks;
+
+  // Array<{files: Array<(filename: string)>, name?: string}>
+  let {chunks} = compilation;
+
+  if (includeNames || excludeNames) {
+    // Only include chunks in config.chunks and
+    // exclude any chunks in config.excludeChunks
+    chunks = filterChunks(chunks, includeNames, excludeNames);
```

Comment:
Would it be accurate to do the following:

Rename `chunks` to `allChunks` or `allCompilationChunks`, renaming `includeNames` to `chunksToInclude` and `excludeNames` to `chunksToExclude` and then add a variables `chunksToPrecache` would would result in:

```javascript
let chunksToPrecache = allChunks;
if (chunksToInclude || chunksToExclude) {
    chunksToPrecache = filterChunks(allChunks, chunksToInclude, chunksToExclude);
}
```